File: MiniDOMBase/DMD/Anagrafica/Persone/Events/PersonaEventArgs.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
    

    Public Class PersonaEventArgs
        Inherits System.EventArgs

        Private m_Persona As CPersona

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Public Sub New(ByVal persona As CPersona)
            Me.New()
            Me.m_Persona = persona
        End Sub

        Public ReadOnly Property Persona As CPersona
            Get
                Return Me.m_Persona
            End Get
        End Property

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class
 

End Class

File: MiniDOMBase/DMD/Anagrafica/Persone/Events/PersonaEventArgs.cs
//TODO rimuovere PersonaEventArgs da Javascript

//using System;

//namespace minidom
//{
//    public partial class Anagrafica
//    {

//        /// <summary>
//        /// Evento relativo ad una persona fisica o giuridica
//        /// </summary>
//        [Serializable]
//        public class PersonaEventArgs 
//            : EventArgs
//        {
            
//            [NonSerialized] private CPersona m_Persona;

//            /// <summary>
//            /// Csotruttore
//            /// </summary>
//            public PersonaEventArgs()
//            {
//                ////DMDObject.IncreaseCounter(this);
//            }

//            /// <summary>
//            /// Csotruttore
//            /// </summary>
//            /// <param name="persona"></param>
//            public PersonaEventArgs(CPersona persona) : this()
//            {
//                m_Persona = persona;
//            }

//            /// <summary>
//            /// Restituisce la persona che ha generato l'evento
//            /// </summary>
//            public CPersona Persona
//            {
//                get
//                {
//                    return m_Persona;
//                }
//            }

//            /// <summary>
//            /// Distruttore
//            /// </summary>
//            ~PersonaEventArgs()
//            {
//                ////DMDObject.DecreaseCounter(this);
//            }
//        }
//    }
//}